modules/templates/templates/templates.tsx

```diff
@@ -0,0 +1,250 @@
+import React, { useState, useEffect } from "react";
+import { GlobalCore } from "@/core/module/module.types";
+import { GenesisTemplate } from "@/core/module/core.types";
+import styles from "./styles/templates.module.css";
+import DeleteConfirmation from "@/resources/containers/delete-confirmation";
+import {
+  FaTrash,
+  FaEdit,
+  FaPlus,
+  FaSave,
+  FaRegFolderOpen,
+} from "react-icons/fa";
+import { useTranslation } from "react-i18next";
+import MessageHandler from "@/core/message-handler";
+import { useRouter } from "next/router";
+import Table from "@/resources/table/table";
+import { useService } from "@/core/module/service.factory";
+import Pagination from "@/resources/table/pagination";
+
+const messageHandler = MessageHandler.get();
+
+const Templates = () => {
+  const router = useRouter();
+  const { t } = useTranslation();
+  const templateService = useService("templates");
+  const [templates, setTemplates] = useState<GenesisTemplate[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [templateToDelete, setTemplateToDelete] = useState<number | null>(null);
+  const [editingTemplate, setEditingTemplate] =
+    useState<GenesisTemplate | null>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [totalRecords, setTotalRecords] = useState(0);
+  const [pageSize] = useState(10);
+
+  useEffect(() => {
+    fetchTemplates(currentPage);
+  }, [currentPage]);
+
+  const fetchTemplates = async (page: number) => {
+    setIsLoading(true);
+    try {
+      const response = await templateService?.getTemplates(page, pageSize);
+      if (response) {
+        setTemplates(response.data);
+        setTotalPages(response.totalPages);
+        setTotalRecords(response.totalCount);
+      }
+    } catch (error) {
+      messageHandler.handleError(t("templates.fetchError"));
```

Comment:
No try catch in the component layer, shall be done if required (that most times is not) in the service

Finally is ok xD can have finally wihtout catch.